### PR TITLE
Feature/unit tests for editor

### PR DESCRIPTION
**Pull request by seryj**
Few unit tests for testing the functionality in the editor tab.
Right now, 4 tests out of (new) 8 are failing. 2 of them are related to handling of non-ascii characters. 2 are user-experience-related and are not that important (when in the beginning/end of the line, the cursor should jump to previous/next line when left/right buttons are pressed).

### CODE REVIEW

File: src/app/editor/mod.rs [hunk 1]
```diff
@@ -163,6 +163,14 @@ impl Input {
         Ok(AppReturn::Continue)
     }
 
+    fn number_chars_in_current_line(&self) -> usize {
```
Comment (matthewmturner) [inline src/app/editor/mod.rs:166]: Can you add `#[allow(dead_code)]` and a comment how this is used for non-ascii characters

**Comment by matthewmturner** [comment]
Thank you again @seryj for the contribution!  This is great and a very similar approach as i was going to take for testing the editor.  Just a few things:

- Could you fix the fmt and clippy errors from CI?
- For the 2 tests that are expected to fail can you add `#[should_panic]` and a comment on why
- Would you mind uncommenting the below as well which i had turned off until adding tests:

https://github.com/datafusion-contrib/datafusion-tui/blob/8f9275e14dfd84bd16228035a98d5384ca397579/.github/workflows/rust.yml#L48-L51

I will work on getting the user improvements added!

Thank you again for the contribution!

**Comment by seryj** [comment]
Ah, thanks for the hints. I need to get used to Github a little bit. And I was not aware of clippy yet. I will try to fix the changes tomorrow.

**Comment by seryj** [comment]
Github is kidding me today. :-/ For some reason I cannot push the changes. I will check what's wrong tomorrow since it's already quite late today. Sorry for this delay.

**Comment by matthewmturner** [comment]
@seryj no rush :) I appreciate your contribution

**Comment by matthewmturner** [comment]
@seryj - ive started doing some enhancements that have impacted your PR.  Once I finish on my current change I can just pick up fixing the remaining CI issues on this.  I don't want you to have to have to deal with that.

**Comment by matthewmturner** [comment]
@seryj i think i got this ready to merge.  did you have anything else you wanted to add?

**Comment by seryj** [comment]
No, not in this PR.

**Pull request merged**